tests/snippets/stdlib_os.py

```diff
@@ -58,3 +31,46 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 	os.putenv(ENV_KEY, ENV_VALUE)
 	os.unsetenv(ENV_KEY)
 	assert os.getenv(ENV_KEY) == None
+
+
+if os.name == "nt":
+	assert os.sep == "\\"
+else:
+	assert os.sep == "/"
+
+class TestWithTempDir():
+	def __enter__(self):
+		if os.name == "nt":
+			base_folder = os.environ["TEMP"]
+		else:
+			base_folder = "/tmp"
+		name = base_folder + os.sep + "test_os"
+		os.mkdir(name)
+		self.name = name
+		return name
+
+	def __exit__(self, exc_type, exc_val, exc_tb):
+		for f in os.listdir(self.name):
+			# Currently don't support dir delete.
+			os.remove(self.name + os.sep + f)
+		os.rmdir(self.name)
```

Comment:
I changed it to not delete the folder. This fails only on rustpython and not in Cpython so I will try to solve this at a later stage. The test folder is in temp and very small so that should not be a problem.